src/tools/storage-key-recipe-resolver.ts

```diff
@@ -172,3 +182,13 @@ export function findLongRunningArcId(recipe: Recipe): string | null {
   }
   return null;
 }
+
+
+/** Determines if create handle is volatile. */
+function isVolatileCreateHandle(handle: Handle): boolean {
+  if (handle.capabilities.isEmpty()) return true;
+  if (!handle.capabilities.isPersistent &&
```

Comment:
Question for the reviewer: Can handles be volatiles and queryable?

Reply:
not at the moment, but we are rethinking capabilities, so in the future stores will be able to be in-memory and queryable.